### PR TITLE
Part 2

### PR DESCRIPTION
**Pull request by dariiabutorina**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public Movie create(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
```
Comment (andromash) [inline src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]: we don't need to set it manually, Hibernate does it by itself

File: src/main/java/com/dev/cinema/dao/GenericDao.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.dev.cinema.dao;
+
+import java.util.List;
+
+public interface GenericDao<T> {
```
Comment (andromash) [inline src/main/java/com/dev/cinema/dao/GenericDao.java:5]: Not sure we need to create it in this hw

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        } catch (Exception exception) {
+            throw new DataBaseDataExchangeException(GET_ALL_MESSAGE, exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```
Comment (andromash) [inline src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]: In this case, we don't use transaction so you can put session to resources block

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+public class MovieDaoImplementation implements MovieDao {
+    public static final String CREATE_MESSAGE = "Failed to insert the Movie entity";
+    public static final String GET_ALL_MESSAGE = "Failed to retrieve all of the Movies entities";
+
+    @Override
+    public Movie create(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
```
Comment (frostpv) [inline src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]: трай с ресурсами

File: src/main/java/com/dev/cinema/exceptions/DataBaseDataExchangeException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.dev.cinema.exceptions;
+
+public class DataBaseDataExchangeException extends RuntimeException {
+    public DataBaseDataExchangeException(String message, Throwable cause) {
```
Comment (andromash) [inline src/main/java/com/dev/cinema/exceptions/DataBaseDataExchangeException.java]: not sure if this name of the exception is informative

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
```
Comment (frostpv) [inline src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]: трай с ресурсами

File: src/main/java/com/dev/cinema/model/Movie.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package com.dev.cinema.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Movie {
```
Comment (andromash) [inline src/main/java/com/dev/cinema/model/Movie.java:11]: In this case, Hibernate will create a table "Movie" but it is not the best practice to name the table with the first letter in the Upper case. You can use an annotation Table with parameter name

**Review by andromash — CHANGES_REQUESTED** [review]
Masterpiece!

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java (hunk 4 above)
Comment (dariiabutorina) [reply src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]: Didn't work with such construction

**Review by andromash — APPROVED** [review]
Incredible! Amazing! 10 out of 10!

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.21.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.17</version>
+        </dependency>
+    </dependencies>
```
Comment (dmytro-hryhoruk) [inline pom.xml]: let's move dependencies after properties

File: src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import com.dev.cinema.library.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImplementation implements MovieDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/implementation/MovieDaoImplementation.java]:
```suggestion
public class MovieDaoImpl implements MovieDao {
```
More than enough)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**